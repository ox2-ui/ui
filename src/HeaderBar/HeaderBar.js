import React, { PropTypes } from 'react';
import classNames from 'classnames';
import Icon from '@ox2/icon/Icon';
import Button from '@ox2/button/Button';

/**
 * HeaderBar Component
 */
const HeaderBar = (props) => {
  const { className: classNameProp, icon, image, title, buttonHandler, button2Handler, onIconClick, buttonLabel, button2Label } = props;

  const className = classNames(
    'display:flex flex:column flex:0-auto color:white border-color:brand border:bottom border:2',
    classNameProp
  );

  return (
    <div className={className}>
      <div className="display:flex flex:items-center flex:44px padding-x:7">
        {/* Picture */}
        {do {
          if (icon) {
            <Icon size={'25px'} onClick={onIconClick} icon={icon} className="text-color:brand" />;
          } else if (image) {
            <img
              src={image}
              width="30"
              height="30"
              className="border:rounded"
              role="presentation"
            />;
          }
        }}
        {/* Title */}
        <div className="w:100p font:roboto-condensed text-color:neutral-dark size:14 text:left text:truncate text:uppercase padding-x:7">{title}</div>
        {/* Buttons */}
        {do {
          if (button2Handler) {
            <Button btn="header-button small raised" onClick={button2Handler} className="flex:0-auto margin-r:7">{button2Label}</Button>;
          }
        }}
        {do {
          if (buttonHandler) {
            <Button btn="header-button small raised" onClick={buttonHandler} className="flex:0-auto">{buttonLabel}</Button>;
          }
        }}
      </div>
    </div>
  );
};

HeaderBar.propTypes = {
  /**
   * The css class name of the root element.
   */
  className: PropTypes.string,
};

export default HeaderBar;
