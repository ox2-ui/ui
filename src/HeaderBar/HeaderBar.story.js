import React from 'react';
import { storiesOf } from '@kadira/storybook';
import HeaderBar from './HeaderBar';

const handleOnClick = () => console.log('🍇'); // eslint-disable-line no-console


storiesOf('HeaderBar', module)
  .add('with image', () => (
    <HeaderBar
      title="Linda Sanders"
      image="https://randomuser.me/api/portraits/women/21.jpg"
    />
  ))
  .add('with icon', () => (
    <HeaderBar
      title="Cactus settings"
      icon="cactus"
    />
  ))
  .add('with button', () => (
    <HeaderBar
      title="Cactus settings"
      icon="cactus"
      buttonHandler={handleOnClick}
      buttonLabel={'Filter'}
    />
  ));
