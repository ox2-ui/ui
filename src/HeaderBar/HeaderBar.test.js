/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import HeaderBar from './HeaderBar';

const handleOnClick = () => console.log('🍇'); // eslint-disable-line no-console


it('renders correctly', () => {
  const wrapper = mount(
    <HeaderBar
      title="Cactus settings"
      icon="cactus"
      buttonHandler={handleOnClick}
      buttonLabel={'Filter'}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
