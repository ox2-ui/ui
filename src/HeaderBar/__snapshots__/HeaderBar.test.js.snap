exports[`test renders correctly 1`] = `
<HeaderBar
  buttonHandler={[Function]}
  buttonLabel="Filter"
  icon="cactus"
  title="Cactus settings">
  <div
    className="display:flex flex:column flex:0-auto color:white border-color:brand border:bottom border:2">
    <div
      className="display:flex flex:items-center flex:44px padding-x:7">
      <Icon
        className="text-color:brand"
        icon="cactus"
        size="25px">
        <span
          className="icon-cactus text-color:brand"
          style={
            Object {
              "color": undefined,
              "fontSize": "25px",
            }
          } />
      </Icon>
      <div
        className="w:100p font:roboto-condensed text-color:neutral-dark size:14 text:left text:truncate text:uppercase padding-x:7">
        Cactus settings
      </div>
      <Button
        btn="header-button small raised"
        className="flex:0-auto"
        onClick={[Function]}>
        <button
          className="flex:0-auto"
          data-btn="header-button small raised"
          onClick={[Function]}>
          Filter
        </button>
      </Button>
    </div>
  </div>
</HeaderBar>
`;
